pulser/sampler/sampler.py

```diff
@@ -61,12 +64,10 @@ def sample(
 
         ch_noises = list(common_noises)
 
+        slm_on = len(seq._slm_mask_targets) > 0 and len(seq._slm_mask_time) > 0
```

Comment:
Empty lists evaluate to `False`, so you can just do:
```suggestion
        slm_on = seq._slm_mask_targets and seq._slm_mask_time
```

Reply:
Thanks for this nice tip